scality_sproxyd_client/sproxyd_client.py

```diff
@@ -254,3 +260,57 @@ def handle_200_or_206(response):
         }
 
         return self._do_http('get_object', handlers, 'GET', name, headers)
+
+    def put_object(self, name, body, user_metadata=None):
+        """Connect to sproxyd and put an object."""
+
+        # The Content-Length is automatically set to the correct value.
```

Comment:
Body is sent all the way done to `httplib.HTTPConnection.send`

```
        blocksize = 8192
        if hasattr(data,'read') and not isinstance(data, array):
            if self.debuglevel > 0: print "sendIng a read()able"
            datablock = data.read(blocksize)
            while datablock:
                self.sock.sendall(datablock)
                datablock = data.read(blocksize)
        else:
            self.sock.sendall(data)
```

I can test it but it seems safe.